src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -948,6 +949,14 @@ public virtual SecurityToken ReadToken(string token)
         /// <param name="validationParameters">A <see cref="TokenValidationParameters"/>  required for validation.</param>
         /// <returns>A <see cref="TokenValidationResult"/></returns>
         public virtual TokenValidationResult ValidateToken(string token, TokenValidationParameters validationParameters)
+        {
+            var validationResult = ValidateTokenAsync(token, validationParameters);
+            validationResult.Wait();
```

Comment:
>validationResult.Wait(); [](http://example.com/codeflow?start=12&length=24)

For my benefit can you explain why you went with Wait() here?

I don't have a lot of experience around best practices with mixing async and synchronous methods but some searching around makes this look like it might not be ideal.

[https://stackoverflow.com/questions/9343594/how-to-call-asynchronous-method-from-synchronous-method-in-c](https://stackoverflow.com/questions/9343594/how-to-call-asynchronous-method-from-synchronous-method-in-c)

For one thing, as I understand, this will wrap all exceptions into an AggregateException which I think would be a behavior change here, is that desired?